packages/forms/test/form_group_spec.ts

```diff
@@ -1825,6 +1825,386 @@ describe('FormGroup', () => {
+         fakeAsync(() => {
+           const logs: string[] = [];
+           const c =
+               new FormControl('', null, simpleAsyncValidator({timeout: 1, shouldFail: true}));
+
+           attachEventsLogger(c, logs);
+
+           expect(logs.length).toBe(0);
+
+           c.setValue('new!', {emitEvent: true});
+
+           tick(1);
+
+           // The `setValue` call invoked synchronously cancels the initial run of the
+           // `asyncValidator` (which would cause the control status to be changed to `INVALID`), so
+           // the log contains only events after calling `setValue`.
+           expect(logs).toEqual([
+             'value: "new!"',    // value change emitted by `setValue`
+             'status: PENDING',  // status change emitted by `setValue`
+             'status: INVALID'   // async validator run after `setValue` call
+           ]);
+         }));
+
+      it('should cancel initial run of the async validator and not emit anything', fakeAsync(() => {
+           const logger: string[] = [];
+           const c =
+               new FormControl('', null, simpleAsyncValidator({timeout: 1, shouldFail: true}));
+
+           attachEventsLogger(c, logger);
+
+           expect(logger.length).toBe(0);
+
+           c.setValue('new!', {emitEvent: false});
+
+           tick(1);
+
+           // Because we are calling `setValue` with `emitEvent: false`, nothins is emitted
+           // and our logger remains empty
+           expect(logger).toEqual([]);
+         }));
+
+      it('should run the sync validator on stand alone controls and set status to `INVALID`',
+         fakeAsync(() => {
+           const logs: string[] = [];
+           const c = new FormControl('new!', Validators.required);
+
+           attachEventsLogger(c, logs);
+
+           expect(logs.length).toBe(0);
+
+           tick(1);
+
+           c.setValue('', {emitEvent: true});
+
+           tick(1);
+
+           expect(logs).toEqual([
+             'value: ""',       // value change emitted by `setValue`
+             'status: INVALID'  // status change emitted by `setValue`
+           ]);
+         }));
+
+      it('should run the sync validator on stand alone controls and set status to `VALID`',
+         fakeAsync(() => {
+           const logs: string[] = [];
+           const c = new FormControl('', Validators.required);
+
+           attachEventsLogger(c, logs);
+
+           expect(logs.length).toBe(0);
+
+           tick(1);
+
+           c.setValue('new!', {emitEvent: true});
+
+           tick(1);
+
+           expect(logs).toEqual([
+             'value: "new!"',  // value change emitted by `setValue`
+             'status: VALID'   // status change emitted by `setValue`
+           ]);
+         }));
+    });
+
+    describe('combination of multiple form controls', () => {
+      it('should run the async validator on FormGroup and FormControl and set status to `VALID`',
```

Comment:
I don't think we run validators for the `FormGroup` in that test (please correct me if I'm wrong).
```suggestion
      it('should run the async validator on the FormControl  added to the FormGroup and set status to `VALID`',
```

Reply:
Good catch.